Sources/System/FilePath/FilePathComponents.swift

```diff
@@ -72,6 +72,16 @@ extension FilePath {
   }
 }
 
+extension FilePath.Component {
+  public static func == (lhs: Self, rhs: Self) -> Bool {
```

Comment:
Unfortunately, this would be a newly exported symbol, so it needs to come with an availability declaration.

To clean up the mess that resulted from adding `public` members to an `internal` type, we should add this implementation in a form that is back-deployable.